src/build/analyzer.ts

```diff
@@ -49,7 +49,7 @@ export class StreamAnalyzer extends Transform {
       this.allFragments.push(shell);
     }
 
-    if (entrypoint) {
+    if (entrypoint && !shell && fragments.length === 0) {
```

Comment:
This means that the entry point must absolutely only import shell? Are we enforcing that throughout our tooling?